lib/rules/no-import-of-scoped-modules-during-ssr.js

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { docUrl } = require('../util/doc-url');
+
+const formFactorModule = '@salesforce/client/formFactor';
+const featureFlagModule = '@salesforce/featureFlag';
+
+const userScopedModulesSet = new Set([
+    '@salesforce/user/Id',
+    '@salesforce/user/isGuest',
+    '@salesforce/userPermission',
+    '@salesforce/customPermission',
+]);
+
+function isUserScopedModule(importSource) {
+    return [...userScopedModulesSet].some((module) => importSource.startsWith(module));
+}
+
+module.exports = {
+    meta: {
+        type: 'problem',
+        docs: {
+            description:
+                'Disallow static imports of user-scoped modules, issue warnings for dynamic imports, and discourage use of formFactor.',
+            category: 'LWC',
+            url: docUrl('no-import-of-scoped-modules-during-ssr'),
+            recommended: true,
+        },
+        messages: {
+            noStaticUserScopedImport:
+                'Static import of @salesforce user modules is not allowed in server-side rendered components.',
+            noDynamicUserScopedImport:
+                'Dynamic import of user-scoped modules is not allowed in server-side rendered components. User-scoped values should either be retrieved from a specific cookie or derived from the existing SID cookie.',
```

Comment:
@khangsfdc What is the recommendation here? A warning like this?

```suggestion
            noDynamicUserScopedImport:
                'Dynamic import of user-scoped modules is discouraged in server-side components.',
```

Reply:
I think the dynamic import warning should not be specific to scoped modules. The warning for dynamic imports should just warn the customer that it does not affect SSR output.

Reply:
Ok, so there should be a warning for **all** dynamic imports during SSR. Something like this:

> Dynamic imports are not processed on the server and do not affect server-side rendered output.